pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -3883,8 +3963,15 @@ func (r *ClickHouseReader) GetLatencyMetricMetadata(ctx context.Context, metricN
 	}, nil
 }
 
-func isColumn(tableStatement, field string) bool {
-	return strings.Contains(tableStatement, fmt.Sprintf("`%s` ", field))
+func isColumn(tableStatement, attrType, field, datType string) bool {
+	// value of attrType will be `resource` or `tag`, if `tag` change it to `attribute`
+	prefix := attrType
+	if attrType == string(v3.AttributeKeyTypeTag) {
+		prefix = constants.Attributes[:len(constants.Attributes)-1]
+	}
+	name := fmt.Sprintf("%s_%s_%s", strings.ToLower(prefix), field, strings.ToLower(datType))
```

Comment:
Shouldn't the `strings.ToLower(datType)` come in the middle? Can you use this `isColumn` and the new `prepareColumn`? elsewhere in this PR to avoid potentially missing updating in multiple places?

Reply:
Yeah makes sense.

Reply:
Have made the changes